.dev/test/testImageManager.php

```diff
@@ -0,0 +1,114 @@
+$path_rel = '/img/p/';
+$image_types = ImageType::getImagesTypes('products', true);
+foreach ($settings as $setting) {
+    foreach ($setting as $name => $value) {
+        Configuration::updateValue($name, $value);
+    }
+    $first_line = true;
+    foreach ($extensions as $extension){
+        echo '<tr' . ($first_line ? ' class= "first" ' : '') . '><td>';
+        $first_line = false;
+        echo 'Extension: '. $extension;
+        foreach ($setting as $name => $value) {
+            echo  '<br>' . $name . ': '. $value;
+        }
+        echo '</td>';
+        $imageFolder = Image::getImgFolderStatic($id_image);
+        $files = glob($path_abs . $imageFolder . '*');
+        foreach($files as $file){
+            if(is_file($file) || is_link($file)) {
+                unlink($file);
+            }
+        }
+        chmod($path_abs . $imageFolder, 0775);
+        copy(
+            __DIR__ . DIRECTORY_SEPARATOR . 'fixtures' . DIRECTORY_SEPARATOR . 'testImageManager.' . $extension,
+            $path_abs . $imageFolder . $id_image . '.jpg'
+        );
+        foreach ($image_types as $image_type) {
+            $thumb = $path_abs . $imageFolder . $id_image . '-' . $image_type['name'] . '.jpg';
+            ImageManager::resize(
+                $path_abs . $imageFolder . $id_image . '.jpg',
+                $thumb,
+                (int) $image_type['width'],
+                (int) $image_type['height']
+            );
+            echo '<td><img src="' . $path_rel . $imageFolder .  $id_image . '-' . $image_type['name'] . '.jpg' . '"></td>';
+            echo '<td class="infos">' . 'Name: ' . $id_image . '-' . $image_type['name'] . '.jpg';
+            foreach (getInfo($thumb) as $name => $value) {
+                echo '<br>' . ucfirst($name) . ': '. $value;
+            }
+            echo '</td>';
+
+        }
+        echo '</tr>';
+        if($setting['PS_IMAGE_QUALITY'] == 'webp_fb'){
+            echo '<tr><td>Duplicates in webp format</td>';
+            foreach ($image_types as $image_type) {
+                $thumb = $path_abs . $imageFolder . $id_image . '-' . $image_type['name'] . '.webp';
+                echo '<td><img src="' . $path_rel . $imageFolder . $id_image . '-' . $image_type['name'] . '.webp' . '"></td>';
+                echo '<td class="infos">' . 'Name: ' . $id_image . '-' . $image_type['name'] . '.webp';
+                foreach (getInfo($thumb) as $name => $value) {
+                    echo '<br>' . ucfirst($name) . ': ' . $value;
+                }
+                echo '</td>';
+            }
+            echo '</tr>';
+        }
+        $id_image++;
+    }
+}
+echo '</table></body>';
+
+function getInfo($image) {
+    $size = getimagesize($image);
+    $infos = [];
+    $infos['width'] = $size[0];
+    $infos['height'] = $size[1];
+    $infos['mime'] = $size['mime'];
+    $infos['bytes'] = filesize($image);
+    if(is_link($image)) {
+        $infos['symlink'] = 'true';
+    }
+    return $infos;
+}
```

Comment:
we need to move that into unit tests.

Reply:
I agree, but for the moment I'm not yet able to do it. This implies more than juste to move the files.
Generally speaking, I already wrote unit tests, but for the moment, I dont understand Prestashop's tests yet.
Specifically here, an automatic test could verify the images dimensions in pixel and if it's a symlink or a file.
I don't know how we could test if the images look correctly, i.e. transparency vs. white background, erroneous cropping. Also I cannot expect that the thumbnail generated on an other system is binary equal, I only can expect that it look the same.

Reply:
@boubkerbribri @nesrineabdmouleh Hello guys, please, would you have a minute to move these tests and do it, so it fits the other mechanisms? It's difficult even for otherwise skilled people. Please 👍

This PR is extremely valuable and would a provide the biggest performance for years, and we need time to implement it into classic and theme-refacto. The sooner this gets merged, the better. 🚀

Reply:
Hello @Hlavtox, 
The QA team handles only tests on the directory `tests/UI`. Units tests are not something we are comfortable with.

Reply:
@boubkerbribri Oh, didn't know that man. Sooo, could you ping somebody who could do this, so we can advance with this PR?

Reply:
@PululuK @boubkerbribri 
I'm still waiting on you...